x/programs/v2/cmd/plan.go

```diff
@@ -192,7 +192,7 @@ func (c *runCmd) RunStep(ctx context.Context, db *state.SimpleMutable) (*Respons
 	resp := newResponse(index)
 	err = runStepFunc(ctx, c.log, db, step.Endpoint, step.MaxUnits, step.Method, params, step.Require, resp)
 	if err != nil {
-		c.log.Debug("simulation step", zap.Error(err))
+		c.log.Error("simulation step err", zap.Error(err))
```

Comment:
I'm wondering if the simulation should halt if there is an error, both cases kinda make sense

Reply:
I think there may be cases where we want the error to be returned without killing the whole simulator.

Reply:
True, this would be part of https://github.com/ava-labs/hypersdk/issues/595 (I think)

Reply:
Agreed with both!